sky/clouds/aws.py

```diff
@@ -477,51 +480,91 @@ def _get_image_id(
         return image_id_str
 
     @classmethod
-    def get_image_size(cls, image_id: str, region: Optional[str]) -> float:
-        if image_id.startswith('skypilot:'):
-            return DEFAULT_AMI_GB
-        assert region is not None, (image_id, region)
-        # first try the cache
-        workspace_profile = aws.get_workspace_profile()
-        kv_cache_key = f'aws:ami:size:{workspace_profile}:{region}:{image_id}'
-        image_size = kv_cache.get_cache_entry(kv_cache_key)
-        if image_size is not None:
-            return float(image_size)
-        # if not found in cache, query the cloud
+    def _describe_image_with_retry(
+        cls,
+        image_id: str,
+        region: str,
+        log_context: str,
+    ) -> Optional['ec2_type_defs.ImageTypeDef']:
         image_not_found_message = (
-            f'Image {image_id!r} not found in AWS region {region}.\n'
-            f'\nTo find AWS AMI IDs: https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-images.html#examples\n'  # pylint: disable=line-too-long
+            f'Image {image_id!r} not found in AWS region {region} - '
+            f'can\'t get {log_context}.\n\n'
+            f'To find AWS AMI IDs: https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-images.html#examples\n'  # pylint: disable=line-too-long
             'Example: ami-0729d913a335efca7')
         max_retries = 3
+        debug_message = 'no describe_images response'
         for iteration in range(1, max_retries + 1):
             try:
                 client = aws.client('ec2', region_name=region)
-                image_info = client.describe_images(ImageIds=[image_id]).get(
-                    'Images', [])
+                response = client.describe_images(ImageIds=[image_id])
+                metadata = response['ResponseMetadata']
+                debug_message = (
+                    'describe_images response:\n'
+                    f'  status code: {metadata["HTTPStatusCode"]}\n'
+                    f'  retry attempts: {metadata["RetryAttempts"]}\n'
+                    f'  len(images): {len(response["Images"])}\n'
+                    f'  next token: {metadata.get("NextToken")}')
```

Comment:
nit: would any of these cause KeyError?

Reply:
I changed these based on the boto3 type definitions. We use `.get` on fields like NextToken that are NotRequired. I assume those type definitions are reliable but it's technically possible that they are not.

Reply:
I see! Either way sounds fine to me. Just would like to avoid the non-critical logging to block any critical path.